tests/queries/test_db_returning.py

```diff
@@ -0,0 +1,43 @@
+import datetime
+import unittest
+
+from django.db import connection, router
+from django.db.models.sql import InsertQuery
+from django.test import TestCase
+
+from .models import Author, ReturningModel
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', 'PostgreSQL tests')
+class ReturningValuesTests(TestCase):
+
+    @staticmethod
+    def get_sql(model):
+        db = router.db_for_write(model)
+        query = InsertQuery(model)
+        query.insert_values(model._meta.fields, [], raw=False)
+        compiler = query.get_compiler(using=db)
+        compiler.return_id = True
+        return compiler.as_sql()[0][0]
+
+    def test_pk_only(self):
+        self.assertIn('RETURNING "queries_author"."id"', self.get_sql(Author))
+
+    def test_multiple_returning_fields(self):
+        self.assertIn(
+            'RETURNING "queries_returningmodel"."id", "queries_returningmodel"."created"',
+            self.get_sql(ReturningModel),
+        )
+
+    def test_insert(self):
+        obj = ReturningModel.objects.create()
+        self.assertTrue(obj.pk)
+        self.assertIsInstance(obj.created, datetime.datetime)
+
+    def test_bulk_insert(self):
+        objs = [ReturningModel() for _ in range(10)]
+        ReturningModel.objects.bulk_create(objs)
+        for obj in objs:
+            with self.subTest(obj=obj):
+                self.assertTrue(obj.pk)
+                self.assertIsInstance(obj.created, datetime.datetime)
```

Comment:
@codingjoe I'm going to check why `test_insert` and `test_bulk_insert` fail on Oracle with the current implementation :thinking: .